dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/deduplication/hibernate/HibernatePotentialDuplicateStore.java

```diff
@@ -291,15 +283,22 @@ public void moveEnrollments(
         duplicate.getEnrollments().stream()
             .filter(e -> !e.isDeleted())
             .filter(e -> enrollments.contains(e.getUid()))
-            .collect(Collectors.toList());
+            .toList();
 
     enrollmentList.forEach(duplicate.getEnrollments()::remove);
 
+    User currentUser =
+        CurrentUserUtil.getCurrentUserDetails() == null
+            ? null
+            : entityManager.getReference(
+                User.class, CurrentUserUtil.getCurrentUserDetails().getId());
+
```

Comment:
is the ternary check needed here?
would `entityManager.getReference(User.class, CurrentUserUtil.getCurrentUserDetails().getId())` suffice? returning the valid User or null if not found?